pkg/packetbrokeragent/grpc_gspba.go

```diff
@@ -159,12 +160,25 @@ func (s *gsPbaServer) UpdateGateway(ctx context.Context, req *ttnpb.UpdatePacket
 
 	if ttnpb.HasAnyField(req.FieldMask.GetPaths(), "contact_info") {
 		adminContact, techContact := toPBContactInfo(req.Gateway.GetContactInfo())
-		updateReq.AdministrativeContact = &packetbroker.ContactInfoValue{
-			Value: adminContact,
+
+		// Only replaces the administrative and technical contacts if no error happens.
```

Comment:
It's possible that the user is deleted after it is set as the contact, and in these cases it will be 'not found', while still being referenced in the contact fields.

As you have said, there are many reasons unrelated to the business logic itself that can cause this call to fail - connection issues, database issues, and others. While the caller cannot fix them, it should at least log them with a warning. You can consider skipping the logging for 'not found entities'.